notification_service/tests/test_cors.py

```diff
@@ -0,0 +1,97 @@
+import logging
+from unittest.mock import patch
+
+import pytest
+import sentry_sdk
+from django.test import Client
+
+from notification_service.cors_utils import check_cors_setting
+
+
+@pytest.mark.django_db
+def test_cors_allowed_origin(settings, live_server, client: Client):
+    settings.CORS_ALLOWED_ORIGINS = [
+        live_server.url,
+    ]
+    settings.CORS_ORIGIN_ALLOW_ALL = False
+    response = client.get("/admin/login/", HTTP_ORIGIN=live_server.url)
+    assert response.status_code == 200
+    assert response["Access-Control-Allow-Origin"] == live_server.url
+
+
+@pytest.mark.django_db
+def test_cors_disallowed_origin(settings, live_server, client: Client):
+    settings.CORS_ALLOWED_ORIGINS = [live_server.url]
+    settings.CORS_ORIGIN_ALLOW_ALL = False
+    response = client.get("/admin/login/", HTTP_ORIGIN="http://malicious.com")
+    assert response.status_code == 200  # The request still succeeds...
+    assert (
+        "Access-Control-Allow-Origin" not in response
+    )  # ...but CORS headers are not present
```

Comment:
Does this mean that the `get` operation did not succeed even though it gave 200?

Reply:
Yes, it is the browser that chooses to reject the response, because there seems to be something fishy there. The browser is asked to download something that the server does not trust in. The API server gives the approved origins as a hint in response headers.